src/file.jl

```diff
@@ -56,32 +56,39 @@ function h5open(filename::AbstractString, mode::AbstractString = "r";
     try
         pv = setproperties!(fapl, fcpl; pv...)
         isempty(pv) || error("invalid keyword options $pv")
-        file = h5open(filename, mode, fapl, fcpl; swmr=swmr)
-        return file
+        return h5open(filename, mode, fapl, fcpl; swmr=swmr)
     finally
         close(fapl)
         close(fcpl)
     end
 end
 
-
 """
-    function h5open(f::Function, args...; swmr=false, pv...)
+    function h5open(f::Function, args...; pv...)
 
 Apply the function f to the result of `h5open(args...; kwargs...)` and close the resulting
-`HDF5.File` upon completion. For example with a `do` block:
+`HDF5.File` upon completion.
+For example with a `do` block:
 
     h5open("foo.h5","w") do h5
         h5["foo"]=[1,2,3]
     end
 
 """
-function h5open(f::Function, args...; swmr=false, pv...)
-    file = h5open(args...; swmr=swmr, pv...)
-    try
-        f(file)
-    finally
-        close(file)
+function h5open(f::Function, args...; context = copy(CONTEXT), pv...)
+    file = h5open(args...; pv...)
+    task_local_storage(:hdf5_context, context) do
+        ctx = task_local_storage(:hdf5_context)
```

Comment:
This is kinda confusing why do you need to use `ctx` instead of the passed in `context` ?

Reply:
If I understood `tls` correctly, this construction allow access of the context from withing a task (private). Anyway, this was a request of @mkitti to use tls so I will let him handle it.